### ISSUE TITLE
It looks very fun~~

### ISSUE THREAD

**Issue by HeMinzhang** [zodiacon/Win10SysProgBookSamples#1]
It looks very fun, whether it is windows system driver development or kernel programming related?

Do you have a trial chapter?

**Comment by zer0cat** [comment]
This book is very good and very useful
please but this book and you will not regret
Book is very good, you can write drivers in VC, not in old batch files.

**Comment by zer0cat** [comment]
Pavel, what about this book? https://www.amazon.com/Windows-via-C-6th/dp/0135953391/
I correctly understand that that book will not be never. And will there be only ebook, from learnpub?
In all cases , your books is very good. Nobody wants to write book for C++ and WinApi.

**Comment by zodiacon** [comment]
Yes, that book was theoretically planned, but then I decided working with MS press was too annoying in the past, and their royalties are ridiculous, so I decided to go my own way...

**Comment by zer0cat** [comment]
Hello again. Thank you for your book. It is very useful.
Can you add some exercises at the end of each chapter (such as in chapters 1-4 )? This contributes to a better understanding of the material.
In page 405 you have mistake "This means the thread that performs the analysis is **blokced** until the **anaylsis** is complete. "

**Comment by zodiacon** [comment]
I will add exercises, that's a good observation.
And will fix the typos.

**Comment by zer0cat** [comment]
You have mistake in 4 page with content "in this chapter". "NUM**N**A" instead of NUMA.

**Comment by zodiacon** [comment]
Indeed, thanks

**Comment by zer0cat** [comment]
Hello. Book part2, UIPI
...Sending uncontrolled messages to a **winodw** can cause..

**Comment by zodiacon** [comment]
Fixed (with some other typos)
